clients/gitlabrepo/client_test.go

```diff
@@ -69,3 +83,79 @@ func TestCheckRepoInaccessible(t *testing.T) {
 		})
 	}
 }
+
+func TestListCommits(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name         string
+		responsePath string
+		commits      []clients.Commit
+		wantErr      bool
+	}{
+		{
+			name:         "Error in ListRawCommits",
+			responsePath: "./testdata/invalid-commits",
+			commits:      []clients.Commit{},
+			wantErr:      true,
+		},
+		{
+			name:         "No commits in repo",
+			responsePath: "./testdata/valid-checkruns-1", // Contains Empty Array
+			commits:      []clients.Commit{},
+			wantErr:      false,
+		},
+		{
+			name:         "Error in getMergeRequestDetail",
+			responsePath: "./testdata/valid-commits",
+			commits:      []clients.Commit{},
+			wantErr:      true,
```

Comment:
do we know why getMergeRequestDetail is erroring, is it because the goodGatewayRoundTripper isn't returning a body?

I'm not sure this test is essential to the change, so up to you on refining it or cutting it.

Reply:
I think it's failing to validate the url info in the stub repourl we're passing in. Once we get around to making tests for gitlab graphql code we'll need to make a more robust test round tripper. 

I kept it in so I was covering both paths in the check for a non-zero commit array length. If you don't think it needed I'll remove it until we can tackle tests for `getMergeRequestDetail` and `zip`.